### PR TITLE
chore(ci): update workflows

### PR DESCRIPTION
**Pull request by eglitise**
This PR updates the four current GitHub workflows (PR validation / main branch push validation / Crowdin update / Crowdin sync):
* Use only Node ~18~ LTS in all workflows
* Add excluded paths to PR and push validation workflows
* In the PR validation workflows, run linter as a separate job

### CODE REVIEW

File: .github/workflows/crowdin-sync-translations.yml [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v3
-    - name: Use Node.js 16.x
+    - name: Use Node.js 18.x
```
Comment (mykola-mokhnach) [inline .github/workflows/crowdin-sync-translations.yml:15]: simply change it to `lts/*`

File: .github/workflows/crowdin-update-resources.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    - uses: actions/checkout@v3
+    - name: Use Node.js 18.x
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18.x
+        cache: 'npm'
+    - name: Install Dependencies
+      run: npm ci
+    - name: Crowdin Update
+      run: node ./ci-jobs/crowdin-update-resources.js
```
Comment (mykola-mokhnach) [inline .github/workflows/crowdin-update-resources.yml:23]: same as above

File: .github/workflows/pr-validation.yml [hunk 3]
```diff
@@ -1,28 +1,58 @@
...
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18.x
+        cache: 'npm'
+    - name: Install Dependencies
+      run: npm ci
+    - name: Build
+      run: npm run build --if-present
+    - name: Test
+      run: npm run test:unit && npm run test:integration
```
Comment (mykola-mokhnach) [inline .github/workflows/pr-validation.yml:42]: do unit tests need xvfb?
I would also run them in parallel with integration to have better perf

File: .github/workflows/crowdin-sync-translations.yml [hunk 4]
```diff
@@ -1,29 +1,29 @@
-name: Weekly Sync Translations job
+# Retrieves non-English translations from Crowdin and creates a PR with new changes
+
+name: Sync Crowdin Translations
 
 on:
-  workflow_dispatch:
```
Comment (mykola-mokhnach) [inline .github/workflows/crowdin-sync-translations.yml:4]: Why the possibility of custom execution has been removed?

File: .github/workflows/pr-validation.yml (hunk 3 above)
Comment (eglitise) [reply .github/workflows/pr-validation.yml:42]: For now it looks like neither unit nor integration tests require a display. I'm guessing this was meant more for e2e tests.
I'm not sure there's much benefit to running the current test suites in parallel. On my mac mini the unit tests take 350ms, and integration tests take 4s. The main bulk of time in this workflow is spent on the previous build step - perhaps it's better to extract that step?

File: .github/workflows/crowdin-sync-translations.yml (hunk 1 above)
Comment (eglitise) [reply .github/workflows/crowdin-sync-translations.yml]: done

File: .github/workflows/crowdin-sync-translations.yml [hunk 5]
```diff
@@ -1,29 +1,29 @@
...
-    - run: npm ci
-      name: Install dependencies
-    - run: npm run crowdin-sync
-      name: Crowdin Sync
+        node-version: 18.x
+        cache: 'npm'
+    - name: Install Dependencies
+      run: npm ci
+    - name: Crowdin Sync
+      run: node ./ci-jobs/crowdin-sync-translations.js
```
[the remark this answers is not in the document]
Comment (eglitise) [reply .github/workflows/crowdin-sync-translations.yml]: moved the line back

File: .github/workflows/crowdin-update-resources.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js 18.x
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18.x
```
[the remark this answers is not in the document]
Comment (eglitise) [reply .github/workflows/crowdin-update-resources.yml]: done

File: .github/workflows/crowdin-update-resources.yml (hunk 2 above)
Comment (eglitise) [reply .github/workflows/crowdin-update-resources.yml]: moved this to package.json

File: .github/workflows/crowdin-sync-translations.yml (hunk 4 above)
Comment (eglitise) [reply .github/workflows/crowdin-sync-translations.yml:4]: My mistake, I saw the colon in the end and thought that this is an empty placeholder configuration. I've added it back.

File: .github/workflows/pr-validation.yml (hunk 3 above)
Comment (eglitise) [reply .github/workflows/pr-validation.yml:39]: After removing xvfb, the Ubuntu runner job still passes, so this removal appears to be safe.

**Review by mykola-mokhnach — APPROVED** [review]

File: .github/workflows/crowdin-update-resources.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# Updates Crowdin with any changes in the English translation file (/assets/locales/en/translation.json)
+
+name: Update Crowdin English Resources
+
+on:
+  push:
+    branches: [ main ]
```
Comment (mykola-mokhnach) [inline .github/workflows/crowdin-update-resources.yml:7]: how do you think, would it make sense to also add a possibility of manual execution to this workflow, so one could push resources from a branch?

**Review by jlipps — APPROVED** [review]

File: .github/workflows/crowdin-update-resources.yml (hunk 7 above)
Comment (eglitise) [reply .github/workflows/crowdin-update-resources.yml:7]: I think the sync translations job only pulls non-English translations, so if Crowdin gets manually pushed resources that aren't on main, then the next sync translations job would only add the translated strings for all languages except English. Not sure how the app would react there.

**Pull request merged**